Python/bltinmodule.c

```diff
@@ -2819,29 +2836,44 @@ zip_new(PyTypeObject *type, PyObject *args, PyObject *kwds)
         PyTuple_SET_ITEM(ittuple, i, it);
     }
 
-    /* create a result holder */
-    result = PyTuple_New(tuplesize);
-    if (result == NULL) {
-        Py_DECREF(ittuple);
-        return NULL;
+    return zip_from_ittuple(type, ittuple, tuplesize, strict);
+}
+
+static PyObject *
+zip_vectorcall(PyObject *type, PyObject * const*args,
+               size_t nargsf, PyObject *kwnames)
+{
+    Py_ssize_t nargs = PyVectorcall_NARGS(nargsf);
+    Py_ssize_t nkwargs = (kwnames) ? PyTuple_GET_SIZE(kwnames) : 0;
+    int strict = 0;
+
+    if (nkwargs == 1) {
```

Comment:
Would it be possible to fallback on zip_new() is there are keywords?

Just optimize zip(arg) positional-only API.